### PR TITLE
fix: make addSigner idempotent by resuming pending operations on retry

### PR DESCRIPTION
**Pull request by albertoelias-crossmint**
## Description

Fixes [WAL-9383](https://linear.app/crossmint/issue/WAL-9383/addsigner-retry-shouldnt-fail).

When `addSigner` is called and the approval step (step 2) fails mid-flight (network error, timeout, etc.), the signer is left in a pending state on the backend. On retry, `addSigner` would blindly call `registerSigner` again, which fails because the signer already exists.

This PR makes `addSigner` idempotent by checking for existing signer state before registration:
- **Already approved** → return immediately (no-op)
- **Pending operation** → resume the pending approval (transaction or signature) instead of re-registering
- **Failed / not found** → proceed with fresh registration as before

The approval logic (handling transactions for Solana/Stellar and signatures for EVM) is extracted into a shared `completeSignerRegistration` helper, used by both the fresh registration path and the resume path. This reuses the existing `getSignerState` / `getPendingSignerOperation` infrastructure that `recover()` already relies on.

### Things for reviewer to check
- Every `addSigner` call now makes an extra `getSigner` API call upfront to check existing state — confirm this latency tradeoff is acceptable
- Verify `isApprovedSignerStatus` (checks for `"success"` / `"active"`) covers all terminal approved states
- Race condition edge case: if a signer transitions from pending → approved between the upfront check and the approval attempt, the approval call should still be safe (idempotent on the backend), but worth confirming

## Test plan

- 6 new unit tests covering addSigner retry/idempotency:
  - Resume pending EVM signature instead of re-registering
  - Resume pending Solana transaction instead of re-registering
  - Early return when signer is already approved
  - True idempotency: calling `addSigner` twice with the same signer yields the same result (only one `registerSigner` call)
  - `prepareOnly` mode returns operation ID when resuming a pending operation
  - Fall through to fresh registration when signer is in failed state
- 3 existing `recover()` tests updated to account for the additional `getSigner` call from `addSigner`
- All 293 tests pass locally, lint passes

## Package updates

- `@crossmint/wallets-sdk`: patch bump (changeset added)

Link to Devin session: https://crossmint.devinenterprise.com/sessions/408fbed01afa4dae9266b753cae9bc78
Requested by: @albertoelias-crossmint

---

<a href="https://crossmint.devinenterprise.com/review/crossmint/crossmint-sdk/pull/1782" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

File: packages/wallets/src/wallets/wallet.ts [hunk 1]
```diff
@@ -680,70 +707,73 @@ export class Wallet<C extends Chain> {
...
+        options?: AddSignerOptions
+    ): Promise<AddSignerReturnType<C> | WalletSigner> {
+        if (options?.prepareOnly) {
+            if (pendingOperation == null) {
+                return { ...signer } as AddSignerReturnType<C>;
+            }
+            const operationId =
+                pendingOperation.type === "transaction"
+                    ? { transactionId: pendingOperation.id }
+                    : { signatureId: pendingOperation.id };
```
Comment (panosinthezone) [inline packages/wallets/src/wallets/wallet.ts]: Lets not default here and process explicitly for the type

File: packages/wallets/src/wallets/wallet.ts [hunk 2]
```diff
@@ -680,70 +707,73 @@ export class Wallet<C extends Chain> {
...
+            return { ...signer, ...operationId } as AddSignerReturnType<C>;
+        }
 
-                return { ...registeredSigner, status: "success" as const } as WalletSigner;
+        if (pendingOperation != null) {
+            if (pendingOperation.type === "transaction") {
+                await this.approveTransactionAndWait(pendingOperation.id);
+            } else {
+                await this.approveSignatureAndWait(pendingOperation.id);
             }
```
Comment (panosinthezone) [inline packages/wallets/src/wallets/wallet.ts]: same here

**Review by panosinthezone — APPROVED** [review]

File: packages/wallets/src/wallets/wallet.ts (hunk 1 above)
Comment (albertoelias-crossmint) [reply packages/wallets/src/wallets/wallet.ts]: @devinai lets take a look. if you were to change it, what would be the cleanest and simplest approach?

File: packages/wallets/src/wallets/wallet.ts (hunk 2 above)
Comment (albertoelias-crossmint) [reply packages/wallets/src/wallets/wallet.ts]: @devinai lets take a look. if you were to change it, what would be the cleanest and simplest approach?

File: packages/wallets/src/wallets/wallet.ts (hunk 1 above)
Comment (albertoelias-crossmint) [reply packages/wallets/src/wallets/wallet.ts]: @devinai do it
